### PR TITLE
Fixes for ffcv integration

### PR DESCRIPTION
**Pull request by dskhudia**
These are fixes we need based on full model test + performance 

Distributed flag for ffcv loader and fix the performance issue of ffcv __len__ function. 

~~WIP: Not ready for review yet.~~

### CODE REVIEW

**Comment by dskhudia** [comment]
@ravi-mosaicml This is ready for review.

**Review by ravi-mosaicml — APPROVED** [review]
LGTM, though can we add a test case for using the FFCV loader? Thinking something along the lines of we write a synthetic dataset then train mnist-cifar10 one epoch on it. Feel free to do this in another PR (I'll create a JIRA to track it)

**Comment by dskhudia** [comment]
> LGTM, though can we add a test case for using the FFCV loader? Thinking something along the lines of we write a synthetic dataset then train mnist-cifar10 one epoch on it. Feel free to do this in another PR (I'll create a JIRA to track it)

Sure. Will add it. Let me do it in the next PR though.

File: composer/datasets/ffcv_utils.py [hunk 1]
```diff
@@ -10,7 +10,35 @@
...
+    if not ffcv_installed:
+        raise ImportError(
+            textwrap.dedent("""\
+            Composer was installed without ffcv support.
+            To use ffcv with Composer, please install ffcv in your environment."""))
+
+
+def ffcv_monkey_patches():
+    _require_ffcv()
+
```
[the remark this answers is not in the document]
Comment (dskhudia) [reply composer/datasets/ffcv_utils.py:30]: Added.

File: composer/datasets/ffcv_utils.py [hunk 2]
```diff
@@ -10,7 +10,35 @@
...
+try:
+    import ffcv  # type: ignore
+    ffcv_installed = True
+except ImportError:
+    ffcv_installed = False
+
+
+def _require_ffcv():
+    if not ffcv_installed:
+        raise ImportError(
```
[the remark this answers is not in the document]
Comment (dskhudia) [reply composer/datasets/ffcv_utils.py]: Done. However, ffcv has no conda package. In the import error, we also point out using pip to install to it should be fine.

**Pull request merged**